packages/typescript/src/typescript.ts

```diff
@@ -31,35 +31,37 @@ export function typescript(configFilePath: string, extraCompilerOptions: ts.Comp
     const { compilerOptions } = config;
     const basePath = path.dirname(absPath);
 
-    const fullCompilerOptions = {
+    const fullCompilerOptions: ts.CompilerOptions = {
       ...compilerOptions,
+      tsBuildInfoFile: path.join(basePath, '.betterer.tsbuildinfo'),
       ...extraCompilerOptions
     };
     config.compilerOptions = fullCompilerOptions;
 
-    const host = ts.createCompilerHost(fullCompilerOptions);
-    const configHost = {
+    const host = ts.createIncrementalCompilerHost(fullCompilerOptions, ts.sys);
+    const configHost: ts.ParseConfigHost = {
       ...host,
       readDirectory: ts.sys.readDirectory.bind(ts.sys),
       useCaseSensitiveFileNames: host.useCaseSensitiveFileNames()
     };
     const parsed = ts.parseJsonConfigFileContent(config, configHost, basePath);
 
     const rootNames = await resolver.validate(parsed.fileNames);
-    const program = ts.createProgram({
+    const program = ts.createIncrementalProgram({
       ...parsed,
       rootNames,
       host
     });
 
     const { diagnostics } = program.emit();
 
-    const preEmitDiagnostic = ts.getPreEmitDiagnostics(program);
```

Comment:
Doesn't support the return value of `createIncrementalProgram` so the individual diagnostics are just spread into the big array instead